src/flags.rs

```diff
@@ -58,6 +59,12 @@ pub fn set_flags(
   opts.optflag("", "v8-options", "Print V8 command line options.");
   opts.optflag("", "types", "Print runtime TypeScript declarations.");
 
+  // set proper parsing style - it allows to pass arguments after script name (and do parsing
+  // in userland)
+  //     deno --allow-env foo.ts --foo=bar
+  // parsing will stop when `foo.ts` is encountered
+  opts.parsing_style(ParsingStyle::StopAtFirstFree);
```

Comment:
Hrm.. I'd rather not StopAtFirstFree. It would be nice to intermix flags from core and user programs.

Reply:
Maybe you can show a test case of what's going wrong?

Reply:
Apologies if I'm way out of line here since I'm just starting to get involved with deno and I might be misreading this comment.

Wouldn't intermixing flags from core and user programs cause problems down the line?

It looks like a clear separation is the only way to defend programs (and core) from potential issues that might arise from arguments being added / removed, and would avoid confusion?

example: `deno program.ts --version` is this passing `--version` to program? or to deno? both? 
example2: `deno program.ts --some-arg` initially `--some-arg` is not a `deno` argument, but down the line, it's aded to `deno`, would the program stop working?

or did I get it wrong?